pkg/cache/clusterqueue.go

```diff
@@ -225,7 +227,9 @@ func (c *ClusterQueue) updateResourceGroups(in []kueue.ResourceGroup) {
 			rg.Flavors = append(rg.Flavors, fQuotas)
 		}
 	}
-	c.AllocatableResourceGeneration++
+	if !reflect.DeepEqual(oldRG, c.ResourceGroups) {
```

Comment:
One consideration. I'm wondering if we should decide which `DeepEqual` we should use. `reflect` pkg vs `equality.Semantic` pkg.

However, I think this is out of the scope of this PR.

Reply:
we are using equality everywhere else, so we should stick with it.

Reply:
I could see the case for using `reflect` package here:

https://github.com/kubernetes-sigs/kueue/blob/1ecd79f748af2617111d0bfa1ca848f9e172f4d7/pkg/config/config.go#L119-L129

Reply:
However, I agree with using `equality` everywhere.